scripts/create-deployment-kpt.sh

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env bash
+# Copyright 2024 The Nephio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Stricter error handling
+set -e # Exit on error
+set -u # Must predefine variables
+set -o pipefail # Check errors in piped commands
+
+function error() {
+  cat <<EOF
+Error: ${1}
+Usage: ${0} [flags]
+Supported Flags:
+  --destination DIRECTORY             ... directory in which the Porch kpt pkg will be downloaded to
+  --server-image IMAGE                ... address of the Porch server image
+  --controllers-image IMAGE           ... address of the Porch controllers image
+  --function-image IMAGE              ... address of the Porch function runtime image
+  --wrapper-server-image IMAGE        ... address of the Porch function wrapper server image
+  --enabled-reconcilers RECONCILERS   ... comma-separated list of reconcilers that should be enabled in porch controller
+EOF
+  exit 1
+}
+
+# Flag variables
+DESTINATION=""
+IMAGE_REPO=""
+IMAGE_TAG=""
+ENABLED_RECONCILERS=""
+
+while [[ $# -gt 0 ]]; do
+  key="${1}"
+  case "${key}" in
+    --destination)
+      DESTINATION="${2}"
+      shift 2
+    ;;
+
+   --server-image)
+      SERVER_IMAGE="${2}"
+      shift 2
+    ;;
+
+    --controllers-image)
+      CONTROLLERS_IMAGE="${2}"
+      shift 2
+    ;;
+
+    --function-image)
+      FUNCTION_IMAGE="${2}"
+      shift 2
+    ;;
+
+    --wrapper-server-image)
+      WRAPPER_SERVER_IMAGE="${2}"
+      shift 2
+    ;;
+
+    --enabled-reconcilers)
+      ENABLED_RECONCILERS="${2}"
+      shift 2
+    ;;
+    
+    --kind-context)
+      KIND_CONTEXT_NAME="${2}"
+      shift 2
+    ;;
+
+    *)
+      error "Invalid argument: ${key}"
+    ;;
+  esac
+done
+
+
+function validate() {
+  [ -n "${DESTINATION}"       ] || error "--destination is required"
+  [ -n "${SERVER_IMAGE}"      ] || error "--server-image is required"
+  [ -n "${CONTROLLERS_IMAGE}" ] || error "--controllers-image is required"
+  [ -n "${FUNCTION_IMAGE}"    ] || error "--function-image is required"
+  [ -n "${WRAPPER_SERVER_IMAGE}"    ] || error "--wrapper-server-image is required"
+}
+
+
+function customize-pkg-images {
+	kpt fn eval "${DESTINATION}" --image gcr.io/kpt-fn/search-replace:v0.2.0 -- by-value-regex="${1}" put-value="${2}"
+}
+
+function customize-container-env {
```

Comment:
You could delete this function, see the alternative yq implementation below.

Reply:
done